holoviews/core/spaces.py

```diff
@@ -97,6 +98,56 @@ def _dimension_keys(self):
                 for k in self.keys()]
 
 
+    def _dynamic_mul(self, dimensions, other, keys):
+        """
```

Comment:
Glad to see this functionality but sadly it makes ``__mul__`` into even more of a monster. I've always felt that ``__mul__`` should be implemented once as a separate utility/operation one day. If you agree, I can file an issue to suggest this.